test/epochtalk_server_web/controllers/thread_test.exs

```diff
@@ -181,4 +201,556 @@ defmodule Test.EpochtalkServerWeb.Controllers.Thread do
+      conn: conn
+    } do
+      response =
+        conn
+        |> delete(Routes.thread_path(conn, :purge, -1), %{})
+        |> json_response(400)
+
+      assert response["error"] == "Bad Request"
+      assert response["message"] == "Error, cannot purge thread"
+    end
+
+    @tag authenticated: :mod
+    test "given admin thread and insufficient priority, throws forbidden read error", %{
+      conn: conn,
+      admin_thread: %{post: %{thread_id: thread_id}}
+    } do
+      response =
+        conn
+        |> delete(Routes.thread_path(conn, :purge, thread_id), %{})
+        |> json_response(403)
+
+      assert response["error"] == "Forbidden"
+      assert response["message"] == "Unauthorized, you do not have permission to read"
+    end
+
+    @tag authenticated: :admin
+    test "given super admin thread and insufficient priority, throws forbidden write error", %{
+      conn: conn,
+      super_admin_thread: %{post: %{thread_id: thread_id}}
+    } do
+      response =
+        conn
+        |> delete(Routes.thread_path(conn, :purge, thread_id), %{})
+        |> json_response(403)
+
+      assert response["error"] == "Forbidden"
+      assert response["message"] == "Unauthorized, you do not have permission to write"
+    end
+
+    @tag authenticated: :banned
+    test "given banned authenticated user, throws InvalidPermission forbidden error", %{
+      conn: conn,
+      thread: %{post: %{thread_id: thread_id}}
+    } do
+      assert_raise InvalidPermission,
+                   ~r/^Forbidden, invalid permissions to perform this action/,
+                   fn ->
+                     delete(conn, Routes.thread_path(conn, :purge, thread_id), %{})
+                   end
+    end
+
+    @tag authenticated: :global_mod
+    test "given admin created thread and insufficient priority, throws forbidden error", %{
+      conn: conn,
+      admin_created_thread: %{post: %{thread_id: thread_id}}
+    } do
+      response =
+        conn
+        |> delete(Routes.thread_path(conn, :purge, thread_id), %{})
+        |> json_response(403)
+
+      assert response["error"] == "Forbidden"
+
+      assert response["message"] ==
+               "Unauthorized, you do not have permission to modify another user's thread"
+    end
+
+    @tag :authenticated
+    test "given thread and user who does not own the thread, does not purge thread", %{
+      conn: conn,
+      thread: %{post: %{thread_id: thread_id}}
+    } do
+      assert_raise InvalidPermission,
+                   ~r/^Forbidden, invalid permissions to perform this action/,
+                   fn ->
+                     delete(conn, Routes.thread_path(conn, :purge, thread_id), %{})
+                   end
+    end
+
+    @tag authenticated: :global_mod
+    test "given thread that authenticated user moderates, does purge thread", %{
+      conn: conn,
+      thread: %{post: %{thread_id: thread_id}},
+      user: %{id: user_id}
```

Comment:
https://github.com/epochtalk/epochtalk_server/pull/98#pullrequestreview-2226144176

in response to omitting user from setup return;
you can use this instead:

```users: %{ user: %{id: user_id}}```

Reply:
Is this better than just keeping and returning user for any reason?